planner/core/task.go

```diff
@@ -1597,16 +1597,18 @@ func (p *PhysicalHashAgg) attach2TaskForMpp(tasks ...task) task {
 			return invalidTask
 		}
 		attachPlan2Task(partialAgg, mpp)
-		items := finalAgg.(*PhysicalHashAgg).GroupByItems
-		partitionCols := make([]*expression.Column, 0, len(items))
-		for _, expr := range items {
-			col, ok := expr.(*expression.Column)
-			if !ok {
-				return invalidTask
+		if len(p.MppPartitionCols) == 0 {
```

Comment:
"attach2TaskForMpp" will be called mutiple times. Everytime the group-by columns will have a unique ID. So please construct it everytime.

Reply:
Why the uniqueID will change every time?